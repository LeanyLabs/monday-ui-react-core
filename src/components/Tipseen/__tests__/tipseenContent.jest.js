import React from "react";
import { fireEvent, render, cleanup } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import TipseenContent from "../TipseenContent";
import renderer from "react-test-renderer";
import { DISMISS_BUTTON_TEXT, SUBMIT_BUTTON_TEXT, TIPSEEN_CLOSE_BUTTON_TEST_ID } from "../TipseenConstants";

jest.useFakeTimers();
const renderComponent = ({ ...props }) => {
  return render(<TipseenContent {...props}>content</TipseenContent>);
};

describe("Tipseen content tests", () => {
  describe("Snapshot Tests", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <TipseenContent title="title" isDismissHidden={false}>
            content
          </TipseenContent>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly with empty props", () => {
      const tree = renderer.create(<TipseenContent />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly without title", () => {
      const tree = renderer.create(<TipseenContent>content</TipseenContent>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly without dismiss button", () => {
      const tree = renderer.create(<TipseenContent isDismissHidden>content</TipseenContent>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly without submit button", () => {
      const tree = renderer
        .create(
          <TipseenContent children="content" isSubmitHidden>
            content
          </TipseenContent>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly when override submit button text", () => {
      const tree = renderer
        .create(<TipseenContent submitButtonText="submit button text">content</TipseenContent>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("renders correctly when override dismiss button text", () => {
      const tree = renderer
        .create(<TipseenContent dismissButtonText="dismiss button text">content</TipseenContent>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Integration Tests", () => {
    afterEach(() => {
      cleanup();
    });

    it("call onDismiss function when click on dismiss button", () => {
      const onDismissMock = jest.fn();
      const onDismissButtonProps = {
        onClick: onDismissMock
      };

      const tipseenContent = renderComponent({
        isDismissHidden: false,
        onDismiss: onDismissMock
      });
      const dismissButton = tipseenContent.getByText(DISMISS_BUTTON_TEXT);

      act(() => {
        fireEvent.click(dismissButton);
      });
      expect(onDismissMock.mock.calls.length).toBe(1);
    });

    it("call onSubmit function when click on dismiss button", () => {
      const onSubmitMock = jest.fn();
      const tipseenContent = renderComponent({
        onSubmit: onSubmitMock
      });
      const submitButton = tipseenContent.getByText(SUBMIT_BUTTON_TEXT);

      act(() => {
        fireEvent.click(submitButton);
      });
      expect(onSubmitMock.mock.calls.length).toBe(1);
    });
  });
});
