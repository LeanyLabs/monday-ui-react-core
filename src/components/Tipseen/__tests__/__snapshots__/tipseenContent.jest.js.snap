// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tipseen content tests Snapshot Tests renders correctly 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <span
    aria-level="3"
    className="monday-style-tipseen-title"
    role="heading"
  >
    title
  </span>
  <span
    className="monday-style-tipseen-content_content"
  >
    content
  </span>
  <div
    className="monday-style-tipseen-content_buttons"
  >
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_dismiss monday-style-button monday-style-button--size-small monday-style-button--kind-tertiary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Dismiss
    </button>
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_submit monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`Tipseen content tests Snapshot Tests renders correctly when override dismiss button text 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <span
    className="monday-style-tipseen-content_content"
  >
    content
  </span>
  <div
    className="monday-style-tipseen-content_buttons"
  >
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_submit monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`Tipseen content tests Snapshot Tests renders correctly when override submit button text 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <span
    className="monday-style-tipseen-content_content"
  >
    content
  </span>
  <div
    className="monday-style-tipseen-content_buttons"
  >
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_submit monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      submit button text
    </button>
  </div>
</div>
`;

exports[`Tipseen content tests Snapshot Tests renders correctly with empty props 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <div
    className="monday-style-tipseen-content_buttons"
  >
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_submit monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`Tipseen content tests Snapshot Tests renders correctly without dismiss button 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <span
    className="monday-style-tipseen-content_content"
  >
    content
  </span>
  <div
    className="monday-style-tipseen-content_buttons"
  >
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_submit monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`Tipseen content tests Snapshot Tests renders correctly without submit button 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <span
    className="monday-style-tipseen-content_content"
  >
    content
  </span>
  <div
    className="monday-style-tipseen-content_buttons"
  />
</div>
`;

exports[`Tipseen content tests Snapshot Tests renders correctly without title 1`] = `
<div
  className="monday-style-tipseen-basic-content monday-style-tipseen-content"
>
  <span
    className="monday-style-tipseen-content_content"
  >
    content
  </span>
  <div
    className="monday-style-tipseen-content_buttons"
  >
    <button
      aria-disabled={false}
      className="monday-style-tipseen-content_submit monday-style-button monday-style-button--size-small monday-style-button--kind-primary monday-style-button--color-on-primary-color"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;
